includes/wikia/nirvana/WikiaView.class.php

```diff
@@ -114,30 +109,22 @@ protected function buildTemplatePath( $controllerClass, $methodName, $forceRebui
 			}
 
 			// Service classes must be dispatched by full name otherwise we default to a controller.
-			$controllerBaseName = $app->getBaseName( $controllerClass );
-			if ($app->isService($controllerClass)) {
-				$controllerClass = $app->getServiceClassName( $controllerBaseName );
-			} else {
-				$controllerClass = $app->getControllerClassName( $controllerBaseName );
-			}
-
-			if( empty( $wgAutoloadClasses[$controllerClass] ) ) {
-				throw new WikiaException( "Invalid controller or service name: {$controllerClass}" );
-			}
-
-			// First we look for BaseName_MethodName
```

Comment:
Put these changed lines into a loop looking at each fallback option.  The options are generated by ```getTemplateOptions```